File: signalr-best-practice-client/src/core/interfaces/todotask/todotask.ts
import { ToDoTaskStatusEnum, EntityStatusEnum } from 'src/core/models/enum';
import { IUser } from '../user/user';

export interface ITodotask{
    id: string,
    title: string,
    description: string,
    expiration_date: Date,
    progress_status: ToDoTaskStatusEnum,
    from_user_id: string;
    to_user_id: string;
    created: string;
    updated: string;
    status: EntityStatusEnum;
    from_user: IUser;
    to_user: IUser;
}

File: signalr-best-practice-backend/signalr-best-practice-core/Entities/ToDoTask.cs
using signalr_best_practice_api_models;
using signalr_best_practice_core.Entities.Base;
using signalr_best_practice_core.Entities.UserAccount;
using System;

namespace signalr_best_practice_core.Entities
{
    public class ToDoTask : BaseManualEntity<string>
    {
        public ToDoTaskStatusEnum ProgressStatus { get; set; }
        public string Description { get; set; }

        public DateTime? ExpirationDate { get; set; }

        public string FromUserId { get; set; }
        public virtual User FromUser { get; set; }

        public string ToUserId { get; set; }
        public virtual User ToUser { get; set; }
    }
}
